README.md

```diff
@@ -1,102 +1,269 @@
-![](https://raw.githubusercontent.com/xi-editor/druid/screenshots/images/slider.png)
+[![calc.rs example](https://raw.githubusercontent.com/xi-editor/druid/screenshots/images/calc.png)](./druid/examples/calc.rs)
+[![custom_widget.rs example](https://raw.githubusercontent.com/xi-editor/druid/screenshots/images/custom_widget.png)](./druid/examples/custom_widget.rs)
+[![hello.rs example](https://raw.githubusercontent.com/xi-editor/druid/screenshots/images/hello.png)](./druid/examples/hello.rs)
+[![slider.rs example](https://raw.githubusercontent.com/xi-editor/druid/screenshots/images/slider.png)](./druid/examples/slider.rs)
 
-## Build
+## Concepts
 
-#### Windows
+### druid-shell
 
-run `cargo build`
+The druid toolkit uses druid-shell for a platform-abstracting application shell.
+Druid-shell is responsible for starting a native platform runloop, listening to
+events, converting them into a platform-agnostic representation, and calling a
+user-provided handler with them.
 
-#### macOS
+While druid-shell is being developed with the druid toolkit in mind, it is
+intended to be general enough that it could be reused by other projects
+interested in experimenting with Rust GUI. The druid-shell crate includes a
+couple [non-druid examples].
 
-On macOS, druid requires cairo; see [gtk-rs dependencies] for installation instructions.
+### piet
 
-You may also need to set your `PKG_CONFIG_PATH`; assuming you have installed `cairo` through homebrew, you can build with,
+Druid relies on the [piet 2D graphics abstraction] for drawing. Piet, in turn,
+has multiple backends. Currently: "piet-direct2d", "piet-cairo", and "piet-web".
+Currently the macOS and Linux versions of Druid use the piet-cairo backend,
+while Windows uses piet-direct2d. A GPU backend is planned. Here's a simple
+drawing example from the [custom_widget] example.
 
- ```shell
-$> PKG_CONFIG_PATH="/usr/local/opt/libffi/lib/pkgconfig" cargo build
- ```
-#### Linux
+```rust
+// Create an arbitrary bezier path
+// (base_state.size() returns the size of the layout rect we're painting in)
+let mut path = BezPath::new();
+path.move_to(Point::ORIGIN);
+path.quad_to(
+    (80.0, 90.0),
+    (base_state.size().width, base_state.size().height),
+);
+// Create a color
+let stroke_color = Color::rgb8(0x00, 0x80, 0x00);
+// Stroke the path with thickness 1.0
+paint_ctx.stroke(path, &stroke_color, 1.0);
+
+// Rectangles: the path for practical people
+let rect = Rect::from_origin_size((10., 10.), (100., 100.));
+// Note the Color:rgba8 which includes an alpha channel (7F in this case)
+let fill_color = Color::rgba8(0x00, 0x00, 0x00, 0x7F);
+paint_ctx.fill(rect, &fill_color);
+```
+
+### widgets
+
+Widgets in druid are objects which implement the `Widget` trait. The trait is
+parametrized by a type (`T`) for associated data. All trait methods are provided
+with access to this data, and in the case of `event` the reference is mutable,
```

Comment:
I'm not very familiar with druid, so I'm not sure exactly what `event` refers to here; it seems extra special, which makes me want to be able to click on a link.

Reply:
hmmm... I see `event` and `update` in the example, but I don't know how easy it is to draw the connection.

Reply:
All trait methods (`paint`, `layout`, `event`, and `update`) are provided
with access to this data, and in the case of `event` the reference is mutable,
so that events can directly update the data.

Reply:
Ah, I see. That's definitely easier for me to follow.